plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/local-pickup-select/index.tsx

```diff
@@ -47,7 +51,11 @@ export const LocalPickupSelect = ( {
 				highlightChecked={ true }
 				selected={ selectedOption }
 				options={ pickupLocations.map( ( location ) =>
-					renderPickupLocation( location, packageCount )
+					renderPickupLocation(
+						location,
+						packageCount,
+						selectedOption === location.rate_id
```

Comment:
What if we pass selectedOption here, and we let the function decide if it's selected in the client? 

 I think that would make for a cleaner API.

Reply:
Nice suggestion, thanks, added in [25ca5e7](https://github.com/woocommerce/woocommerce/pull/61852/commits/25ca5e7e55db594df206f47c48b163a2a6160774)